drivers/dma/dma_silabs_ldma.c

```diff
@@ -0,0 +1,564 @@
+		xfer_config->ldmaCfgDstIncSign = ldmaCfgSrcIncSignPos;
+		break;
+	case DMA_ADDR_ADJ_DECREMENT:
+		xfer_config->ldmaCfgDstIncSign = ldmaCfgSrcIncSignNeg;
+		break;
+	case DMA_ADDR_ADJ_NO_CHANGE:
+		xfer_config->ldmaCfgDstIncSign = ldmaCfgSrcIncSignPos;
+		break;
+	default:
+		LOG_ERR("Addr Adjustement error. %d", config->head_block->source_addr_adj);
+		break;
+	}
+
+	switch (config->head_block->dest_addr_adj) {
+	case DMA_ADDR_ADJ_INCREMENT:
+		xfer_config->ldmaCfgSrcIncSign = ldmaCfgDstIncSignPos;
+		break;
+	case DMA_ADDR_ADJ_DECREMENT:
+		xfer_config->ldmaCfgSrcIncSign = ldmaCfgDstIncSignNeg;
+		break;
+	case DMA_ADDR_ADJ_NO_CHANGE:
+		xfer_config->ldmaCfgSrcIncSign = ldmaCfgDstIncSignPos;
+		break;
+	default:
+		break;
+	}
+
+	xfer_config->ldmaLoopCnt = 0;
+
+	ret = dma_silabs_configure_descriptor(config, data, chan_conf);
+	if (ret) {
+		return ret;
+	}
+
+	atomic_set_bit(data->dma_ctx.atomic, channel);
+
+	return ret;
+}
+
+static int dma_silabs_start(const struct device *dev, uint32_t channel)
+{
+
+	const struct dma_silabs_data *data = dev->data;
+	struct dma_silabs_channel *chan_conf = &data->dma_chan_table[channel];
+
+	if (channel > data->dma_ctx.dma_channels) {
+		return -EINVAL;
+	}
+
+	chan_conf->busy = true;
+
+	LDMA_StartTransfer(channel, &chan_conf->xfer_config, chan_conf->desc);
+
+	return 0;
+}
+
+static int dma_silabs_stop(const struct device *dev, uint32_t channel)
+{
+	const struct dma_silabs_data *data = dev->data;
+	struct dma_silabs_channel *chan = &data->dma_chan_table[channel];
+
+	if (channel > data->dma_ctx.dma_channels) {
+		return -EINVAL;
+	}
+
+	LDMA_StopTransfer(channel);
+
+	chan->busy = false;
+
+	/* Clear the interrupt flag. */
+	LDMA->IF_CLR = BIT(channel);
+
+	return 0;
+}
+
+static int dma_silabs_get_status(const struct device *dev, uint32_t channel,
+				 struct dma_status *status)
+{
+	const struct dma_silabs_data *data = dev->data;
+
+	if (channel > data->dma_ctx.dma_channels) {
+		return -EINVAL;
+	}
+
+	if (!atomic_test_bit(data->dma_ctx.atomic, channel)) {
+		return -EINVAL;
+	}
+
+	status->busy = data->dma_chan_table[channel].busy;
+	status->dir = data->dma_chan_table[channel].dir;
+
+	return 0;
+}
+
+static int dma_silabs_init(const struct device *dev)
+{
+	struct dma_silabs_config *config = (struct dma_silabs_config *)dev->config;
+	struct dma_silabs_data *data = (struct dma_silabs_data *)dev->data;
+	struct dma_silabs_channel *dma_channel;
+	LDMA_Init_t dmaInit = LDMA_INIT_DEFAULT;
+
+	if (!device_is_ready(config->clock_dev)) {
+		LOG_ERR("clock control device not ready");
+		return -ENODEV;
+	}
+
+	for (int i = 0; i < data->dma_ctx.dma_channels; i++) {
+		dma_channel = &data->dma_chan_table[i];
+		dma_channel->id = i;
+		dma_channel->busy = false;
+		dma_channel->dir = MEMORY_TO_MEMORY;
+		dma_channel->complete_callback_en = 1;
+		dma_channel->cb = NULL;
+		dma_channel->user_data = NULL;
+		memset(&dma_channel->xfer_config, 0, sizeof(dma_channel->xfer_config));
+	}
```

Comment:
This loop is probably useless since `dma_silabs_channel_##inst` is already zero'd. The other fields will set set by `dma_silabs_configure()` anyway.